File: SteamDroplerApi.Core/Models/Account.cs
using SteamDroplerApi.Core.Configs;

namespace SteamDroplerApi.Core.Models;

public class Account
{
    public string Name { get;  }
    public AccountConfig Config { get; }
    public AccountRunConfig RunConfig { get; }
    public bool IsPlaying { get; set; }
    public Account(string name, AccountConfig config, AccountRunConfig runConfig)
    {
        Name = name;
        Config = config;
        RunConfig = runConfig;
    }
}

File: SteamDroplerFront/src/app/account/models/account.ts
export interface Account {
  name: string;
  isPlaying: boolean;
  config: AccountConfig;
  runConfig: AccountRunConfig;
}

export interface AccountConfig {
  idleEnable: boolean;
  steamId?: number;
  authType?: string;
}

export interface AccountRunConfig {
  lastLoginErrorTime?: Date;
  loginErrorReason?: string;
  lastErrorTime?: Date;
  errorReason?: string;
  updateTime?: Date;
  ownedApps?: number[];
  notOwnedApps?: number[];
  appsToAdd?: number[];
  packagesToAdd?: number[];
}
